src/eventHandlers/FunctionLoadHandler.ts

```diff
@@ -18,11 +20,19 @@ export class FunctionLoadHandler extends EventHandler<'functionLoadRequest', 'fu
     }
 
     async handleEvent(channel: WorkerChannel, msg: rpc.IFunctionLoadRequest): Promise<rpc.IFunctionLoadResponse> {
+        channel.workerIndexingLocked = true;
+
         const response = this.getDefaultResponse(msg);
 
-        const functionId = nonNullProp(msg, 'functionId');
-        const metadata = nonNullProp(msg, 'metadata');
-        if (!channel.functions[functionId]) {
+        channel.log({
+            message: `Worker ${channel.workerId} received FunctionLoadRequest`,
+            level: LogLevel.Debug,
+            logCategory: LogCategory.System,
+        });
+
+        if (!channel.isUsingWorkerIndexing) {

```

Comment:
I'm slightly worried about this if condition here. Is there a possible scenario in which a user is using the old indexing, but still `registerFunction()` gets called somehow, setting `channel.isUsingWorkerIndexing` to true? It feels safer to keep the old check `!channel.functions[functionId]`

Reply:
The user would have to explicitly call `registerFunction()` either from the core api or v4 of the library, which I can't see happening on accident. Our general position has been that we don't support mixing and matching the new & old models and my change takes a stricter stance to prevent that. It was very helpful as I was debugging the multi-worker scenario and should be helpful moving forward for customers as well. For example, it results in clearer error messages catered to each model when things are working quite right.